ingest/src/SnootyDataSource.ts

```diff
@@ -0,0 +1,147 @@
+import fs from "fs";
+import { createInterface } from "readline";
+import { DataSource } from "./DataSource";
+import { Page } from "./updatePages";
+import { snootyAstToMd } from "./snootyAstToMd";
+
+// These types are what's in the snooty manifest jsonl file.
```

Comment:
for these different types, can you leave brief TSDoc comments explaining what each is?

just based on names, it's not clear to me.